rudder-agent/SOURCES/patches/cfengine/80-add-dumpdatatstate-function.patch

```diff
@@ -0,0 +1,45 @@
+diff -upr cfengine-source/libpromises/evalfunction.c cfengine-source-new/libpromises/evalfunction.c
+--- cfengine-source/libpromises/evalfunction.c	2015-09-06 12:39:13.000000000 +0200
++++ cfengine-source-new/libpromises/evalfunction.c	2016-09-22 16:51:19.566248730 +0200
+@@ -2956,6 +2956,32 @@ static FnCallResult FnCallDatastate(Eval
+     return  (FnCallResult) { FNCALL_SUCCESS, (Rval) { state, RVAL_TYPE_CONTAINER } };
+ }
+ 
++static FnCallResult FnCallDumpDatastate(EvalContext *ctx,
++                                        ARG_UNUSED const Policy *policy,
++                                        ARG_UNUSED const FnCall *fp,
++                                        ARG_UNUSED const Rlist *args)
++{
++	    char *path = RlistScalarValue(args);
++
++	    FILE *fout;
++            if ((fout = fopen(path, "w")) == NULL)
++	    {
++	       Log(LOG_LEVEL_ERR, "Cannot open the destination file '%s'. (fopen: %s)",
```

Comment:
It's useful in error messages to know where this error is coming from. Could you prefix it with "dumpdatastate(): " maybe?